src/main/java/gyro/azure/storage/CloudBlobContainerResource.java

```diff
@@ -150,16 +154,34 @@ public boolean refresh() {
     public void create(GyroUI ui, State state) {
         BlobContainerClient blobContainer = blobContainer();
 
-        blobContainer.create();
-
-        blobContainer = blobContainer();
+        BlobContainerCreateOptions blobContainerCreateOptions = new BlobContainerCreateOptions();
 
-        blobContainer.setAccessPolicy(PublicAccessType.fromString(getPublicAccess()), null);
+        if (getPublicAccess() != null) {
+            blobContainerCreateOptions.setPublicAccessType(PublicAccessType.fromString(getPublicAccess()));
+        }
 
         if (!getMetadata().isEmpty()) {
-            blobContainer.setMetadata(getMetadata());
+            blobContainerCreateOptions.setMetadata(getMetadata());
         }
 
+        // We add this wait to create as sometimes there is a delay for the StorageAccount to actually become public.
+        // This results in a 409 error that gets thrown
+        // Unfortunately th api doesn't really provide a good way to determine if the account is actually public.
+        Wait.atMost(2, TimeUnit.MINUTES)
+            .prompt(false)
+            .checkEvery(10, TimeUnit.SECONDS)
+            .until(() -> {
+                Response<Boolean> response =
+                    blobContainer.createIfNotExistsWithResponse(blobContainerCreateOptions, null, null);
+
+                if (response.getStatusCode() == 409) {
+                    return false;
+
+                } else {
+                    return true;
+                }
+            });
+
```

Comment:
We should look into verifying the storage account is set to `public` and get the error when creating this container with public access to fire the retry logic.

That way any other error bubbles up and we have a direct condition for retrying